core/common/src/main/java/alluxio/AlluxioURI.java

```diff
@@ -499,4 +502,30 @@ public String toString() {
     mUriString = sb.toString();
     return mUriString;
   }
+
+  /**
+   * Computes the hash of this URI, with SHA-256, or MD5, or simple hashCode().
+   *
+   * @return HEX encoded hash string
+   */
+  public String hash() {
```

Comment:
How about putting this in the `alluxio.underfs.FileId` class (probably need to move that to a different package and module). Maybe add a method `FileId.fromUri(AlluxioURI)`.

Reply:
I think it is better to have it as a public method of AlluxioUri.
1. If FileId.fromUri(AlluxioURI) is needed, it can use the AlluxioUri.hash() directly.
2. This method can also be used for other purpose, e.g. hashing a path to different workers from client.